app/src/test/java/de/westnordost/streetcomplete/quests/existence/CheckExistenceTest.kt

```diff
@@ -36,4 +38,61 @@ class CheckExistenceTest {
             StringMapEntryDelete("survey_date", "d"),
         )
     }
+
+    @Test fun `title is constructed correctly, issue #2512`() {
+        val newspaperVendingMachineWithName = questType.getQuestTitle(mapOf(
+            "amenity" to "vending_machine",
+            "vending" to "newspapers",
+            "name" to "Bild",
+        ))
+        assertEquals(newspaperVendingMachineWithName, "Is Bild (vending machine) still here?")
+
+        val newspaperVendingMachineWithBrand = questType.getQuestTitle(mapOf(
+            "amenity" to "vending_machine",
+            "vending" to "newspapers",
+            "brand" to "Abendzeitung",
+        ))
+        assertEquals(newspaperVendingMachineWithBrand, "Is Abendzeitung (vending machine) still here?")
+    }
+
+    @Test fun `title is constructed correctly, issue #2640`() {
+        val postBox = questType.getQuestTitle(mapOf(
+            "amenity" to "post_box",
+            "brand" to "Deutsche Post",
+            "operator" to "Deutsche Post AG",
+            "ref" to "Hauptsmoorstr. 101, 96052 Bamberg",
+        ))
+        assertEquals(postBox, "Is Deutsche Post (post box) still here?")
+    }
+
+    @Test fun `title is constructed correctly, issue #2806`() {
+        val namedBench = questType.getQuestTitle(mapOf(
+            "amenity" to "bench",
+            "name" to "Sergey's Seat",
+            "ref" to "600913",
+            "brand" to "Google",
+            "operator" to "Google RESTful",
+        ))
+        assertEquals(namedBench, "Is Sergey's Seat (bench) still here?")
+
+        val unnamedBench = questType.getQuestTitle(mapOf(
+            "amenity" to "bench",
+        ))
+        assertEquals(unnamedBench, "Is this still here? (bench)")
+    }
+}
+
+private fun CheckExistence.getQuestTitle(tags: Map<String, String>): String {
+    val titleTemplate = when (getTitle(tags)) {
+        R.string.quest_existence_name_title -> "Is %1\$s (%2\$s) still here?"
+        R.string.quest_existence_title -> "Is this still here? (%s)"
+        else -> throw IllegalArgumentException("Unknown title resource")
+    }
+
+    // simplified feature name, instead of looking up iD preset
+    val featureName = tags["amenity"]?.replace('_', ' ')
+
+    val titleArgs = getTitleArgs(tags, lazy { featureName })
+
+    return titleTemplate.format(*titleArgs)
```

Comment:
Oh I see, you reimplemented that function for the test. Well, I still think it would be better to move the whole test to `androidTests`, then you don't need this function too.